File: packages/python/qsu/os/getDiskUsage.py
import os
import shutil
from typing import Optional, Union

from ..math.round import round


def getDiskUsage(path: Optional[str] = None, decimals: int = 1) -> Union[int, float]:
	usage = shutil.disk_usage(path or os.getcwd())

	if not usage.total:
		return 0

	return round(usage.used / usage.total * 100, decimals)

File: packages/javascript/lib/node/os/getDiskUsage.ts
import { cwd } from 'node:process';
import { round } from '../../math/round.js';
import { diskSpace } from './_diskSpace.js';

export async function getDiskUsage(path: string = cwd(), decimals = 1): Promise<number> {
	const { total, used } = await diskSpace(path);

	if (!total) {
		return 0;
	}

	return round((used / total) * 100, decimals);
}
